sor/src/main/java/com/bazaarvoice/emodb/sor/db/astyanax/CqlDataWriterDAO.java

```diff
@@ -217,16 +227,23 @@ private void deleteCompactedDeltas(ByteBuffer rowKey, WriteConsistency consisten
         // delete the old deltas & compaction records
         if (_writeToLegacyDeltaTable) {
             BatchStatement legacyTableStatement = new BatchStatement(BatchStatement.Type.UNLOGGED);
-            for (UUID change : changesToDelete) {
-                legacyTableStatement.add(deleteStatement(placement.getDeltaTableDDL(), rowKey, change, consistencyLevel));
+            for (DeltaClusteringKey change : changesToDelete) {
+                legacyTableStatement.add(deleteStatement(placement.getDeltaTableDDL(), rowKey, change.getChangeId(), consistencyLevel));
```

Comment:
so, still going with the range tombstones in case of writing to legacy, Ok by me for now!!!!

Reply:
yes exactly. If this works, we will expand it